skimage/segmentation/seeds_superpixels.py

```diff
@@ -0,0 +1,273 @@
+    label if doing so will improve the quality of the segmentation.
+    ``block_refine_iters`` is the number of passes over the image that are
+    performed for each level.
+    pixel_refine_iters : after all block-level refinements have been
+    performed, the same procedure is repeated at the pixel level.
+    score_threshold : blocks or pixels will only be re-labelled if doing so
+    will result in an improvement score greater than ``score_threshold``.
+    min_label_area_factor : at the start, the image is divided into square
+    labelled regions. Re-labelling blocks or pixels will only be permitted
+    if doing so does not reduce the area covered by a label below
+    ```initial_label_area*min_label_area_factor```.
+    return_steps: If True, will return the intermediate results of the
+    algorithm; see below.
+
+    Returns
+    -------
+    If ``return_steps`` is False, a 2D label image, with labels starting at 1.
+    If ``return_steps`` is True, a tuple of
+    ```(labels, labels_by_level, index_2d, quantized_colours)```, where
+    ``labels`` is the final label image, ``labels_by_level`` is a list of
+    label images generated at each step, starting at the coarsest level and
+    finishing at the finest, ``index_2d`` is the index image after
+    quantization if ``input_2d`` is RGB or greyscale or just ``input_2d`` if
+    it was an index image, and ``quantized_colours`` are the quantized
+    colours or greyscale values as a 2D array if ``input_2d`` is RGB or
+    greyscale or ``None`` otherwise.
+
+    References
+    ----------
+
+    .. [1] M. van den Bergh, X. Boix, G. Roig, B. de Capitani, L. van Gool..
+           "SEEDS: Superpixels extracted via energy-driven sampling."
+           Proceedings of the European Conference on Computer Vision,
+           pages 13-26, 2012.
+    """
+    assert n_levels >= 1
+    assert_nD(input_2d, [2,3])
+    
+    # Get dimensions of input image
+    h, w = input_2d.shape[:2]
+
+    # Compute the number of superpixels vertically and horizontally
+    n_superpixels_h = int(math.sqrt(float(num_superpixels) * w / h)+0.5)
+    n_superpixels_w = n_superpixels_h * w / h
+
+    # Compute the width and height of the blocks at the finest level
+    block_finest_w_f = float(w) / n_superpixels_w / (2**(n_levels-1))
+    block_finest_h_f = float(h) / n_superpixels_h / (2**(n_levels-1))
+
+    # Reduce n_levels if necessary so that finest block sizes are > 1.0
+    while block_finest_w_f <= 1.0 or block_finest_h_f <= 1.0:
+        n_levels -= 1
+        block_finest_w_f = float(w) / n_superpixels_w / (2**(n_levels-1))
+        block_finest_h_f = float(h) / n_superpixels_h / (2**(n_levels-1))
+    
+    # Compute seed size
+    seed_w = max(int(block_finest_w_f+0.5), 1)
+    seed_h = max(int(block_finest_h_f+0.5), 1)
+
+    # Get an index image; determine what to do.
+    if len(input_2d.shape) == 3 and input_2d.shape[2] == 3:
```

Comment:
We've been working against this kind of "magic". Not too sure what to do in this case, but I'm just flagging it for now. We have been going with the `multichannel` keyword argument asking users to explicitely label their images.